tesseract_environment/test/tesseract_environment_unit.cpp

```diff
@@ -151,6 +221,29 @@ Environment::Ptr getEnvironment(bool register_default_contact_managers = false)
   return env;
 }
 
+template <typename S>
+void runEnvInitFailuresTest()
+{
+  auto env = std::make_shared<Environment>();
+  EXPECT_TRUE(env != nullptr);
+  EXPECT_EQ(0, env->getRevision());
+  EXPECT_FALSE(env->reset());
+  EXPECT_FALSE(env->isInitialized());
+  EXPECT_TRUE(env->clone() != nullptr);
+
+  // Test Empty commands
+  Commands commands;
+  EXPECT_FALSE(env->init<S>(commands));
+  EXPECT_FALSE(env->isInitialized());
+
+  // Test scene graph is not first command
+  commands.clear();
+  auto cmd = std::make_shared<MoveJointCommand>("joint_name", "parent_link");
+  commands.push_back(cmd);
+  EXPECT_FALSE(env->init<S>(commands));
+  EXPECT_FALSE(env->isInitialized());
```

Comment:
Perhaps check that CommandHistory is still size 0?

Reply:
Done.